### PR TITLE
Extract MS Office .docx files as archives #1502

### PR DESCRIPTION
**Pull request by a-tinsmith**
Changes:
- `OfficeDocHandler` now recognises MS Office file types as specified in libmagic: Microsoft Word 2007+, Microsoft Excel 2007+, Microsoft Powerpoint 2007+. For reference see https://github.com/file/file/blob/master/magic/Magdir/msooxml
- New unit test for MS Office extractcode
- Updated unit test for typecode
- Fixed an extra space typo in the temporary directory name prefix: 'scancode-tk-tests-'

### CODE REVIEW

File: src/commoncode/testcase.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def get_temp_dir(self, sub_dir_path=None):
             test_tmp_root_dir = os.path.join(scancode_root_dir, 'tmp')
             # not we add a space in the path for testing path with spaces
             test_run_temp_dir = fileutils.get_temp_dir(
-                base_dir=test_tmp_root_dir, prefix='scancode-tk-tests -')
+                base_dir=test_tmp_root_dir, prefix='scancode-tk-tests-')
```
Comment (pombredanne) [inline src/commoncode/testcase.py]: This is a small space that is there for a reason: to ensure that we run tests on paths that contain at least one space, so do you mind reverting this?

Comment (a-tinsmith) [reply src/commoncode/testcase.py]: Updated. I didn't notice the comment one line above!

File: .gitignore [hunk 2]
```diff
@@ -44,6 +44,7 @@ htmlcov
 .idea
 org.eclipse.core.resources.prefs
 .vscode
+.vs
```
Comment (pombredanne) [inline .gitignore]: Just curious: what would this file be?

Comment (a-tinsmith) [reply .gitignore]: I was using Visual Studio and it created this folder. This is VS internal bookkeeping of the project.

Comment (pombredanne) [reply .gitignore]: thx

**Comment by pombredanne** [comment]
Thank you ++

**Pull request merged**